### ISSUE TITLE
Items are invisible while dragging in ClipsDescendants enabled GuiObjects

### ISSUE THREAD

**Issue by Frexsim** [Frexsim/grid-pack#19]
**Describe the bug**
Item's in ClipsDescendants enabled GuiObjects are invisible.

**Reproduction or/and error message**
1. Create a grid inside a ClipsDescendants GuiObject.
2. Drag an item outside of the grid.

**Expected behavior**
The item should not be invisible while dragging.

**Platform(s)**
All

**Version**
0.2.3

**Comment by Vexx2323** [comment]
Sorry for poor explination
TLDR: make a "fake" duplicate frame with Clips turned off and reparent item when moving and dropping (sync up canvasposition if its a scrollframe)


Just have to reparent items outside of the gui while dragging them, might not be the right places or perfect but the work around i found was in the item module on line 225(ish) before create drop highlight do, 

if self.ItemElement.Parent == (Gui with ClipsDescendants) then
	self.ItemElement.Parent = (Gui the same position and size without ClipsDescendants like a parent or sum idk)
end

and then on line  320 (ish) before the self.PotentialRotation = self.rotation do

if self.ItemElement.Parent == (gui without ClipsDescendants) then
	self.ItemElement.Parent = (original Gui with ClipsDescendants)
end


If the GUI is a scroll frame which it was for me the original Gui with ClipsDescendants would be the scroll frame (with the grid in a frame inside the scrollframe) and the gui without ClipsDescendants has to be another scrollframe in the same position, same size and same canvas size with CipsDescendants turned off and then have the canvas position synced inside a localscript somwhere as such


scrollReal:GetPropertyChangedSignal("CanvasPosition"):Connect(function()
	scrollFake.CanvasPosition = scrollReal.CanvasPosition
end)

(scrollFake being the frame inside the scrollframe without ClipsDescendants and the scrollReal being the original frame inside ScrollFrame with it on)
